### ISSUE TITLE
Do SAlib supports only vector inputs?

### ISSUE THREAD

**Issue by kayakml** [SALib/SALib#233]
The shape of output array of my model which i suppose to use as an argument for SAlib's analyse (SOBOL) does not fit the shape of array which sobol needs.

I have a parameter array which has a shape of (28,13) that is produced by SAlib package's saltelli function.
On the other hand, my model computes a solution at numerous time points, say 200. So in the end I have an output array of (28,200)
I must use this output array as an argument for the SAlib's Sobol function.
But as far as i understand from the example in the SAlib's doc , the input argument of Sobol must have a column array such as (28,)
The thing i want to learn is , do i infer correct conclusion from SAlib's document as SAlib's Sobol function needs only a column array ? or can i feed it with an array of (28,200) ?

PS : actually i tried (28,200) and got some error. But I am not sure this error occurs because  SAlib does not support array shape of (28,200) or my code have an error.  (I first asked this to the stackoverflow but couldn't get any answer. 
https://stackoverflow.com/questions/55634208/does-salib-sensitivity-analysis-package-support-only-one-column-vector-input)

**Comment by willu47** [comment]
Hi @kayakml - you'll need to perform the analysis once for each of the 200 outputs using the 28 samples that you have run through the model. Each of the analysis techniques in SALib computes the sensitivity measures for one output at a time.  What you can do is iterate over the array of 200 outputs and pass the column vector of size 28 to the analyze function.

A sample size of 28 seems very small if you are using Sobol. I would expect a sample size of hundreds or thousands to produce sensible results.

**Comment by kayakml** [comment]
Thanx for the answer @willu47 . But i think i could'nt explain my problem thoroughly. If i follow step by step the SAlib doc's example (https://salib.readthedocs.io/en/latest/basics.html) it will be more explanatory. 
In the SAlib's example:          
     1. the Ishigami model has 3 parameters
     2. saltelli.sample produces a parameter set of (8000,3). 
     3. Ishigami.evaluate produces a vector of (8000,)
On the other hand in my work:
     1. my model has 13 parameters
     2. saltelli.sample produces a parameter set of (28,13)    (by the way sample size is just for demo i will increase it later)
     3. my model produces an array of (28,200)

In other words i have only one output that is defined over  a time range comprised of 200 points. So I am not sure SA for each point make sense.

**Comment by spizwhiz** [comment]
@kayakml maybe using some kind of summary statistic on your time series output would be appropriate?  (e.g. average value over time period, cumulative sum at end of time period, some measure of variance over time period)?  This is what I have done with my models that produce time series output.

**Comment by kayakml** [comment]
this sounds a good idea @spizwhiz , i will try it. or maybe i could do SA for each of the time points and then use descriptive statistics to reach a final SA .

**Comment by spizwhiz** [comment]
@kayakml another thing I have seen in papers is plotting the SA for each parameter as a time series.  this sometimes gives useful information as parameter importance can change over the course of model evaluation
